tests/opensearch/basic.bats

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bats
+
+load "$HELPERS_ROOT/test-helpers.bash"
+
+ensure_variables_set BATS_SUPPORT_ROOT BATS_ASSERT_ROOT BATS_DETIK_ROOT BATS_FILE_ROOT TEST_NAMESPACE FLUENTBIT_IMAGE_TAG OPENSEARCH_IMAGE_TAG
+
+load "$BATS_DETIK_ROOT/utils.bash"
+load "$BATS_DETIK_ROOT/linter.bash"
+load "$BATS_DETIK_ROOT/detik.bash"
+load "$BATS_SUPPORT_ROOT/load.bash"
+load "$BATS_ASSERT_ROOT/load.bash"
+load "$BATS_FILE_ROOT/load.bash"
+
+setup() {
+    echo "recreating namespace $TEST_NAMESPACE"
+    if [[ "${SKIP_DELETE_NAMESPACE:-no}" != "yes" ]]; then
+        run kubectl delete namespace "$TEST_NAMESPACE"
+    fi
+    run kubectl create namespace "$TEST_NAMESPACE"
+}
+
+teardown() {
+    if [[ "${SKIP_DELETE_NAMESPACE:-no}" != "yes" ]]; then
+        run kubectl delete namespace "$TEST_NAMESPACE"
+    fi
+}
+
+# These are required for bats-detik
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAME="kubectl -n $TEST_NAMESPACE"
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAMESPACE="${TEST_NAMESPACE}"
+
+
+@test "test fluent-bit forwards logs to opensearch default index" {
+    helm repo add opensearch https://opensearch-project.github.io/helm-charts/ ||  helm repo add opensearch https://opensearch-project.github.io/helm-charts
+    helm repo add fluent https://fluent.github.io/helm-charts/ || helm repo add fluent https://fluent.github.io/helm-charts
+    helm repo update --fail-on-repo-update-fail
+
+    helm upgrade --install --debug --namespace "$TEST_NAMESPACE" opensearch opensearch/opensearch -f ${BATS_TEST_DIRNAME}/resources/helm/opensearch-basic.yaml --set image.tag=${OPENSEARCH_IMAGE_TAG} --wait
+
+    try "at most 15 times every 2s " \
+        "to find 1 pods named 'opensearch-cluster-master-0' " \
+        "with 'status' being 'running'"
+
+    helm upgrade --install --debug --namespace "$TEST_NAMESPACE" fluent-bit fluent/fluent-bit -f ${BATS_TEST_DIRNAME}/resources/helm/fluentbit-basic.yaml --set image.tag=${FLUENTBIT_IMAGE_TAG} --wait
```

Comment:
Set pull policy so it's not always? Just thinking for reuse in other places where we might have a private image and we can just load it.

Or have a generic helm values file we can use or override, or a variable for extra parameters?

Reply:
@patrick-stephens ok, so the request here is to move the install into a function/helper? or to have a values.yaml with the image settings that we can override? because you can always run --set image.pullPolicy=always

Reply:
Using pull always I think is wrong so yeah we should not use that.

My request is that we have a way of providing custom values though if we use this same CI test in other non-public repos.
Currently there is no way to change these parameters so an option to have a custom values.yaml file we can just pass in would be ideal.

Reply:
Not sure if following here, you can pass a custom file with -f or a --set option, what's the use case you have in mind?

Reply:
I can but how do I without changing the test? 
A new variable would be useful that I can just have the file location in.